.github/workflows/code-checker.yml

```diff
@@ -0,0 +1,47 @@
+name: Code Checker
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  linting_and_type-checking:
+    name: Linting, Formatting and Type checking
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
```

Comment:
Use version v4 instead of v2.
For more details : https://github.com/actions/checkout